src/cmds/scripts/pbs_init.d.in

```diff
@@ -783,26 +783,42 @@ site_mom_cleanup()
 	fi
 }
 
+is_registered()
+{
+	if command -v systemctl >/dev/null 2>&1; then
+		systemctl is-enabled pbs > /dev/null 2>&1
+			return $?
+	elif command -v chkconfig; then
+		chkconfig pbs > /dev/null 2>&1
+		return $?
+	fi
+	return 1
+}
+
 # Check whether system is being booted or not 
 # and also update the time in /var/tmp/pbs_boot_check file  
 # return 0 if system is being booted otherwise return 1
 is_boottime()
 {
-	PYTHON_EXE=${PBS_EXEC}/python/bin/python
-	if [ -z "${PYTHON_EXE}" -o ! -x "${PYTHON_EXE}" ] ; then
-		return 1
-	fi
-	
-	BOOTPYFILE="/var/tmp/pbs_bootcheck.py"
-	BOOTCHECKFILE="/var/tmp/pbs_boot_check"
+	if is_registered
+	then
```

Comment:
Please combine the above two into single line. Alternatively you can check for failure of the function is_registered first and if it fails return from here itself. This way code looks much cleaner.